More delegate methods and improved UX

kkieffer opened the pull request:
First, great library.  Thanks for building it. After incorporating it into my own project I discovered a few things I wanted to change.  Added a few items below:

More delegate methods: (1) When search results return exactly one matching item (2) When user taps on search result in list (3) When user taps on map and geocode returns a result (4) User selects an annotation on the map that was added from a search result.

Map improvement: when user taps on search table entry, hide the table and zoom the map to that location

UX improvements: (1) added a public accessible ratio to control how large the expanded stack can be relative to the height of the view. (2) Don’t popup alert dialog when user clicks on the map and geocode search comes back - instead let delegate decide what to do. (3) Removed code that causes an undesirable behavior: when user scrolls in the table view and hits the top, the table slides out of view.

kkieffer committed:
More delegate methods: (1) When search results return exactly one matching item (2) When user taps on search result in list (3) When user taps on map and geocode returns a result (4) User selects an annotation on the map that was added from a search result.

Map improvement: when user taps on search table entry, hide the table and zoom the map to that location

UX improvements: (1) added a public accessible ratio to control how large the expanded stack can be relative to the height of the view. (2) Don’t popup alert dialog when user clicks on the map and geocode search comes back - instead let delegate decide what to do. (3) Removed code that causes an undesirable behavior: when user scrolls in the table view and hits the top, the table slides out of view.

kkieffer committed:
Update MapKitSearch to swift 5

philip-bui commented on Sources/MapKitSearchViewController.swift:
```diff
@@ -739,11 +783,32 @@ extension MapKitSearchViewController: CLLocationManagerDelegate {
+    
+    @available(*, deprecated, message: "Other protocol functions provide finer detail on user action")
     func mapKitSearch(_ mapKitSearchViewController: MapKitSearchViewController, mapItem: MKMapItem)
+
```
LGTM! Would it make sense to keep the controller / self as the first parameter, just to keep the API similar to Apple's. https://developer.apple.com/documentation/mapkit/mkmapviewdelegate. 

If I did patch this in, the version would have to be major (2.X) because there's some breaking changes (unless you can make the new delegate methods optional).

philip-bui commented on Sources/MapKitSearchViewController.swift:
```diff
@@ -759,3 +824,12 @@ class PlaceAnnotation: NSObject, MKAnnotation {
 }
+
+
+extension CLLocationCoordinate2D: Equatable {
```
Would it make sense to put this into the `Extensions` folder. Just for structure and moving the responsibility to that folder.

kkieffer committed:
Update delegate methods so MapKitSearchViewController comes first in param list, moving extension to ClLocationCooordinate2D to its own file in Extensions

kkieffer commented:
Updated to keep controller/self as first parameter, and moved CLLoationCoordinate2D to Extensions folder as suggested.

Regarding optionals on delegate methods.....could do that but would make the protocol @objc.  Not sure if that would be a breaking change anyway.
